antarest/study/business/model/district_model.py

```diff
@@ -0,0 +1,216 @@
+# Copyright (c) 2025, RTE (https://www.rte-france.com)
+#
+# See AUTHORS.txt
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+from enum import Enum
+from typing import Any, List, MutableMapping, Optional
+
+from pydantic import field_validator
+from pydantic.alias_generators import to_camel
+
+from antarest.core.serde import AntaresBaseModel
+
+
+class DistrictApplyFilter(Enum):
+    add_all = "add-all"
+    remove_all = "remove-all"
+
+
+class DistrictUpdate(AntaresBaseModel):
+    """
+    Represents an update of a district.
+    """
+
+    class Config:
+        alias_generator = to_camel
+        extra = "forbid"
+        populate_by_name = True
+
+        @staticmethod
+        def json_schema_extra(schema: MutableMapping[str, Any]) -> None:
+            schema["example"] = DistrictUpdate(
+                comments="Some comment", areas=["z1", "z2", "z3"], output=True
+            ).model_dump(mode="json")
+
+    @field_validator("areas", mode="before")
+    def validate_areas(cls, areas: Any) -> Optional[List[str]]:
+        if areas is None:
+            return areas
+        return list(set(areas))
+
+    #: Indicates whether this district is used in the output (usually all
+    #: districts are visible, but the user can decide to hide some of them).
+    output: Optional[bool] = None
+    #: User-defined comments.
+    comments: Optional[str] = None
+    #: List of areas that will be grouped in the district.
+    areas: Optional[List[str]] = None
+    #: Base filter for the district.
+    apply_filter: Optional[DistrictApplyFilter] = None
+
+
+class DistrictCreation(AntaresBaseModel):
+    """
+    Represents a creation of a district.
+    """
+
+    class Config:
+        alias_generator = to_camel
+        extra = "forbid"
+        populate_by_name = True
+
+        @staticmethod
+        def json_schema_extra(schema: MutableMapping[str, Any]) -> None:
+            schema["example"] = DistrictCreation(
+                name="My District", comments="", areas=["z1", "z2", "z3"], output=True
+            ).model_dump(mode="json")
+
+    @field_validator("areas", mode="before")
+    def validate_areas(cls, areas: Any) -> Optional[List[str]]:
+        if areas is None:
+            return areas
+        return list(set(areas))
+
+    #: Name of the district (this name is also used as a unique identifier).
+    name: str
+    #: Indicates whether this district is used in the output (usually all
+    #: districts are visible, but the user can decide to hide some of them).
+    output: Optional[bool] = True
```

Comment:
Default values are already defined in the `District` class, we should not duplicate them here (otherwise if one day we change on, we need to think also to change it here).
We should use `None` everywhere instead.